File: app/src/main/java/yagoc/players/UserPlayer.java
package yagoc.players;

import yagoc.board.BoardReader;
import yagoc.board.Move;
import yagoc.pieces.PieceColor;

import java.io.Serializable;

public class UserPlayer implements Player, Serializable {
    private final String name;
    private final PieceColor set;

    public UserPlayer(String name, PieceColor set) {
        this.name = name;
        this.set = set;
    }

    public PieceColor pieceColor() {
        return set;
    }

    @Override
    public Move move(BoardReader board) {
        throw new RuntimeException(name + " cannot move without user input.");
    }

    @Override
    public PlayerType type() {
        return PlayerType.user;
    }

    public String name() {
        return name;
    }

    @Override
    public String toString() {
        return set + "\t" + name() + "\t" + type();
    }
}

File: app/src/main/kotlin/yagoc/players/UserPlayer.kt
package yagoc.players

import yagoc.board.BoardView
import yagoc.board.Move
import yagoc.pieces.PieceColor
import java.io.Serializable

class UserPlayer(private val name: String, private val pieceColor: PieceColor) : Player, Serializable {
    override fun pieceColor(): PieceColor {
        return pieceColor
    }

    override fun move(board: BoardView): Move {
        throw RuntimeException("$name cannot move without user input.")
    }

    override fun type(): PlayerType {
        return PlayerType.User
    }

    override fun name(): String {
        return name
    }

    override fun toString(): String {
        return pieceColor.toString() + "\t" + name() + "\t" + type()
    }
}